mas-foundation/src/main/java/com/ca/mas/foundation/JWKPreLoadListener.java

```diff
@@ -0,0 +1,45 @@
+package com.ca.mas.foundation;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.ca.mas.core.conf.ConfigurationManager;
+import com.ca.mas.core.token.JWTRS256Validator;
+
+public class JWKPreLoadListener implements MASLifecycleListener {
+
+    @Override
+    public void onStarted() {
+
+        if (JWTRS256Validator.getJwks() != null) {
```

Comment:
I would prefer to put all this logic under JWTRS256Validator, so something like this during onStarted()
    @Override
    public void onStarted() {

        JWTRS256Validator validator = new JWTRS256Validator();
        validator.loadJWKS(null);
    }

JWTRS256Validator handles all the logic.